package/endpoint_router/endpoint_router/src/endpoint_router.c

```diff
@@ -238,33 +238,31 @@ static void cache_match_process(const forwarding_rule_t *forwarding_rule,
 
   assert(length >= rule_cache->rule_prefixes->prefix_len);
 
-  switch (filter->action) {
-  case FILTER_ACTION_ACCEPT: {
-
-    if (rule_cache->hash != NULL) {
+  if (rule_cache->hash == NULL) {
```

Comment:
Under our "nanomsg/zeromq" free regime, every port needs to have a some list of forwarding rules, otherwise data would not move anywhere.  This represents the case where a port is configured like this:

```
  - name: SBP_PORT_FILEIO_FIRMWARE
    metric: "sbp/fileio_fw"
    pub_addr: "ipc:///tmp/sbp_port_fileio_firmware.pub"
    sub_addr: "ipc:///tmp/sbp_port_fileio_firmware.sub"
    forwarding_rules:
      - dst_port: SBP_PORT_FIRMWARE
        filters:
          - { action: ACCEPT }
```

The `->hash` part of the `rule_cache` caches all prefixes that are part of the rules, so if a port has no prefixes, then it will have no hash... so, in actuality this `if` can probably be an `assert()`, I don't think we're calling `cache_match_process` in the case that a port has no "prefixes" that are part of the `->hash`.